### PR TITLE
Add more linting

### PR DESCRIPTION
**Pull request by RMacfarlane**
Adds some more rules to the `tslint` config and fixes instances of shadowed variables.

@rebornix @shana are there any other `tslint` rules you think we should add?

### CODE REVIEW

**Comment by lumaxis** [comment]
There is a nice set of additional Microsoft TSLint rules here: https://github.com/Microsoft/tslint-microsoft-contrib
They also include some recommendations.

File: src/common/diffPositionMapping.ts [hunk 1]
```diff
@@ -91,9 +90,9 @@ export function getLastDiffLine(prPatch: string): DiffLine {
 export function getDiffLineByPosition(diffHunks: DiffHunk[], diffLineNumber: number): DiffLine {
 	for (let i = 0; i < diffHunks.length; i++) {
 		let diffHunk = diffHunks[i];
-		for (let i = 0; i < diffHunk.diffLines.length; i++) {
-			if (diffHunk.diffLines[i].positionInHunk === diffLineNumber) {
-				return diffHunk.diffLines[i];
+		for (let j = 0; j < diffHunk.diffLines.length; j++) {
```
Comment (IDisposable) [inline src/common/diffPositionMapping.ts]: 😲

**Comment by shana** [comment]
I'm not very familiar with tslint so I'm not sure what's best, this list of rules looks to match other projects I've seen and it's pretty comprehensive.

I've seen projects extend the `tslint-config-prettier` configuration when using prettier (might be a thing we want to do to not worry about formatting at all). I've also seen projects extending `tslint:recommended`, dunno if this list matches the recommended one.

All in all, looks good! I'm just wondering about that change @IDisposable pointed out...

File: src/common/diffPositionMapping.ts (hunk 1 above)
Comment (rebornix) [reply src/common/diffPositionMapping.ts]: Good catch! Even though `let` ensures that the inner `i` shadows the outer one, we should always avoid confusion like this.

Comment (jongio) [inline src/common/diffPositionMapping.ts]: This is a test when clicking the + sign on the line of code that has an existing comment from GitHub.com

Comment (jongio) [reply src/common/diffPositionMapping.ts]: This is a test when replying to an existing comment on GitHub.com

Comment (jongio) [reply src/common/diffPositionMapping.ts]: This is a test when replying to a comment from VS Code extension.

**Pull request merged**